File: cpp/231-240/Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>res(nums.size(), 1);
        
        for (int i = 1; i < nums.size(); i++) {
            res[i] = res[i-1] * nums[i-1];
        }
        
        int tmp = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            res[i] *= tmp;
            tmp *= nums[i];
        }
        
        return res;
    }
};

File: python/231-240/Product of Array Except Self.py
class Solution:
    # @param {integer[]} nums
    # @return {integer[]}
    def productExceptSelf(self, nums):
        ls = len(nums)
        result = [0]*ls
        result[0] = 1
        tmp = 1
        
        for i in range(1,ls):
            result[i] = result[i-1]*nums[i-1]
        for i in range(ls-2, -1, -1):
            tmp = tmp * nums[i+1]
            result[i] = result[i]*tmp
        
        return result


items = [0,1]
s = Solution()
print s.productExceptSelf(items)